--- a/drivers/platform/x86/intel/int3472/tps68470_board_data.c
+++ b/drivers/platform/x86/intel/int3472/tps68470_board_data.c
@@ -287,6 +287,160 @@
 	},
 };
 
+static const struct tps68470_clk_consumer dell_5290_clk_consumers[] = {
+	{ .consumer_dev_name = "i2c-INT3477:00" },
+	{ .consumer_dev_name = "i2c-INT3479:00" },
+};
+
+static struct regulator_consumer_supply dell_5290_int3477_ana_consumer_supplies[] = {
+	REGULATOR_SUPPLY("avdd", "i2c-INT3477:00"),
+};
+
+static struct regulator_consumer_supply dell_5290_int3477_core_consumer_supplies[] = {
+	REGULATOR_SUPPLY("dvdd", "i2c-INT3477:00"),
+};
+
+static struct regulator_consumer_supply dell_5290_int3477_vio_consumer_supplies[] = {
+	REGULATOR_SUPPLY("dovdd", "i2c-INT3477:00"),
+};
+
+static struct regulator_consumer_supply dell_5290_int3477_vcm_consumer_supplies[] = {
+	REGULATOR_SUPPLY("vcc", "i2c-INT3477:00-VCM"),
+};
+
+static struct regulator_consumer_supply dell_5290_int3479_vsio_consumer_supplies[] = {
+	REGULATOR_SUPPLY("avdd", "i2c-INT3479:00"),
+	REGULATOR_SUPPLY("vsio", "i2c-INT3477:00"),
+};
+
+static struct regulator_consumer_supply dell_5290_int3479_aux1_consumer_supplies[] = {
+	REGULATOR_SUPPLY("dvdd", "i2c-INT3479:00"),
+};
+
+static struct regulator_consumer_supply dell_5290_int3479_aux2_consumer_supplies[] = {
+	REGULATOR_SUPPLY("dovdd", "i2c-INT3479:00"),
+};
+
+static const struct regulator_init_data dell_5290_tps68470_core_reg_init_data = {
+	.constraints = {
+		.min_uV = 1200000,
+		.max_uV = 1200000,
+		.apply_uV = 1,
+		.valid_ops_mask = REGULATOR_CHANGE_STATUS,
+	},
+	.num_consumer_supplies = ARRAY_SIZE(dell_5290_int3477_core_consumer_supplies),
+	.consumer_supplies = dell_5290_int3477_core_consumer_supplies,
+};
+
+static const struct regulator_init_data dell_5290_tps68470_ana_reg_init_data = {
+	.constraints = {
+		.min_uV = 2815200,
+		.max_uV = 2815200,
+		.apply_uV = 1,
+		.valid_ops_mask = REGULATOR_CHANGE_STATUS,
+	},
+	.num_consumer_supplies = ARRAY_SIZE(dell_5290_int3477_ana_consumer_supplies),
+	.consumer_supplies = dell_5290_int3477_ana_consumer_supplies,
+};
+
+static const struct regulator_init_data dell_5290_tps68470_vcm_reg_init_data = {
+	.constraints = {
+		.min_uV = 2815200,
+		.max_uV = 2815200,
+		.apply_uV = 1,
+		.valid_ops_mask = REGULATOR_CHANGE_STATUS,
+	},
+	.num_consumer_supplies = ARRAY_SIZE(dell_5290_int3477_vcm_consumer_supplies),
+	.consumer_supplies = dell_5290_int3477_vcm_consumer_supplies,
+};
+
+static const struct regulator_init_data dell_5290_tps68470_vio_reg_init_data = {
+	.constraints = {
+		.min_uV = 1800600,
+		.max_uV = 1800600,
+		.apply_uV = 1,
+		.valid_ops_mask = REGULATOR_CHANGE_STATUS,
+	},
+	.num_consumer_supplies = ARRAY_SIZE(dell_5290_int3477_vio_consumer_supplies),
+	.consumer_supplies = dell_5290_int3477_vio_consumer_supplies,
+};
+
+static const struct regulator_init_data dell_5290_tps68470_vsio_reg_init_data = {
+	.constraints = {
+		.min_uV = 1800600,
+		.max_uV = 1800600,
+		.apply_uV = 1,
+		.valid_ops_mask = REGULATOR_CHANGE_STATUS,
+		.always_on = 1,
+	},
+	.num_consumer_supplies = ARRAY_SIZE(dell_5290_int3479_vsio_consumer_supplies),
+	.consumer_supplies = dell_5290_int3479_vsio_consumer_supplies,
+};
+
+static const struct regulator_init_data dell_5290_tps68470_aux1_reg_init_data = {
+	.constraints = {
+		.min_uV = 1213200,
+		.max_uV = 1213200,
+		.apply_uV = 1,
+		.valid_ops_mask = REGULATOR_CHANGE_STATUS,
+	},
+	.num_consumer_supplies = ARRAY_SIZE(dell_5290_int3479_aux1_consumer_supplies),
+	.consumer_supplies = dell_5290_int3479_aux1_consumer_supplies,
+};
+
+static const struct regulator_init_data dell_5290_tps68470_aux2_reg_init_data = {
+	.constraints = {
+		.min_uV = 1800600,
+		.max_uV = 1800600,
+		.apply_uV = 1,
+		.valid_ops_mask = REGULATOR_CHANGE_STATUS,
+	},
+	.num_consumer_supplies = ARRAY_SIZE(dell_5290_int3479_aux2_consumer_supplies),
+	.consumer_supplies = dell_5290_int3479_aux2_consumer_supplies,
+};
+
+static const struct tps68470_regulator_platform_data dell_5290_tps68470_pdata = {
+	.reg_init_data = {
+		[TPS68470_CORE] = &dell_5290_tps68470_core_reg_init_data,
+		[TPS68470_ANA]  = &dell_5290_tps68470_ana_reg_init_data,
+		[TPS68470_VCM]  = &dell_5290_tps68470_vcm_reg_init_data,
+		[TPS68470_VIO]  = &dell_5290_tps68470_vio_reg_init_data,
+		[TPS68470_VSIO] = &dell_5290_tps68470_vsio_reg_init_data,
+		[TPS68470_AUX1] = &dell_5290_tps68470_aux1_reg_init_data,
+		[TPS68470_AUX2] = &dell_5290_tps68470_aux2_reg_init_data,
+	},
+};
+
+static struct gpiod_lookup_table dell_5290_int3477_gpios = {
+	.dev_id = "i2c-INT3477:00",
+	.table = {
+		GPIO_LOOKUP("tps68470-gpio", 9, "reset", GPIO_ACTIVE_LOW),
+		GPIO_LOOKUP("tps68470-gpio", 7, "powerdown", GPIO_ACTIVE_LOW),
+		{ }
+	}
+};
+
+static struct gpiod_lookup_table dell_5290_int3479_gpios = {
+	.dev_id = "i2c-INT3479:00",
+	.table = {
+		GPIO_LOOKUP("tps68470-gpio", 3, "reset", GPIO_ACTIVE_LOW),
+		GPIO_LOOKUP("tps68470-gpio", 4, "powerdown", GPIO_ACTIVE_LOW),
+		{ }
+	}
+};
+
+static const struct int3472_tps68470_board_data dell_5290_tps68470_board_data = {
+	.dev_name = "i2c-INT3472:05",
+	.tps68470_regulator_pdata = &dell_5290_tps68470_pdata,
+	.n_clk_consumers = ARRAY_SIZE(dell_5290_clk_consumers),
+	.clk_consumers = dell_5290_clk_consumers,
+	.n_gpiod_lookups = 2,
+	.tps68470_gpio_lookup_tables = {
+		&dell_5290_int3477_gpios,
+		&dell_5290_int3479_gpios,
+	},
+};
+
 static const struct dmi_system_id int3472_tps68470_board_data_table[] = {
 	{
 		.matches = {
@@ -316,5 +470,12 @@
 		},
 		.driver_data = (void *)&dell_7212_tps68470_board_data,
 	},
+	{
+		.matches = {
+			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Dell Inc."),
+			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "Latitude 5290 2-in-1"),
+		},
+		.driver_data = (void *)&dell_5290_tps68470_board_data,
+	},
 	{ }
 };
